### ISSUE TITLE
Confluent Cloud connection crashed/lost when 40101 error`

### ISSUE THREAD

**Issue by javabrett** [confluentinc/vscode#336]
Type: <b>Bug</b>

I deleted a cluster in Confluent Cloud Console UI.  When I return to VS Code extension, it hasn't recognised the deletion yet.  If I then try and add a topic, I get an (expected) error:

```
Error creating topic in "cluster_0": {"error_code":40101,"message":"Authentication failed: 1 extensions are invalid! They are: logicalCluster: CLUSTER_NOT_FOUND"}
```

However, my connection to Confluent Cloud also crashes and the connection entry vanishes.

Extension version: 0.17.1
VS Code version: Code 1.93.1 (Universal) (38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40, 2024-09-11T17:20:05.685Z)
OS version: Darwin x64 23.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (16 x 2300)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 3, 3|
|Memory (System)|32.00GB (0.29GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by flippingbits** [comment]
I can confirm that the extension resets the Confluent Cloud connection when running into the above error. I'll discuss the issue with the team and get back to you as soon as possible.

**Comment by alexeyraspopov** [comment]
Thanks for reporting @javabrett. I got https://github.com/confluentinc/vscode/pull/337 that fixes the issue, the changes should be available in v0.18.0 once it is released.
